File: src/Medium/Leetcode2.java
package Medium;

class ListNode {
    int val;
    ListNode next;
    ListNode() {}
    ListNode(int val) { this.val = val; }
    ListNode(int val, ListNode next) { this.val = val; this.next = next; }
}

//Referred lc approach1
class Leetcode2 {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        int carry=0;
        ListNode ansPtr=new ListNode(-1);
        ListNode ptr=ansPtr;
        while(l1!=null || l2!=null){
            int digit1=l1==null?0:l1.val;
            int digit2=l2==null?0:l2.val;
            l1=l1==null?l1:l1.next;
            l2=l2==null?l2:l2.next;
            int sumTemp=digit1+digit2+carry;
            if(sumTemp>=10){
                carry=1;
            }
            else{
                carry=0;
            }
            sumTemp=sumTemp%10;
            ptr.next=new ListNode(sumTemp);
            ptr=ptr.next;
        }
        if(carry==1){
            ptr.next=new ListNode(carry);
        }
        return ansPtr.next;
    }
}

File: src/Python/Medium/Leetcode2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        ptr1=l1
        ptr2=l2
        carry=0
        headDummy=ListNode(0)
        ptr=headDummy
        while ptr1 is not None or ptr2 is not None:
            val1=0 if ptr1 is None else ptr1.val
            val2=0 if ptr2 is None else ptr2.val
            sumDigit=val1+val2+carry
            carry=1 if sumDigit>=10 else 0
            sumDigit=sumDigit%10
            nextNode=ListNode(sumDigit)
            ptr.next=nextNode
            ptr=ptr.next
            ptr1=ptr1 if ptr1 is None else ptr1.next
            ptr2=ptr2.next if ptr2 is not None else ptr2

        if carry==1:
            ptr.next=ListNode(carry)
        return headDummy.next
